web/src/hooks/useUserLocation.ts

```diff
@@ -43,18 +43,21 @@ export const getUserLocation = async (): Promise<UserLocationType> =>
     )
   })
 
-const useUserLocation = (): { data: UserLocationType | null; refresh: () => void } => {
-  const [refreshState, setRefreshState] = useState(false)
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  const { data } = useLoadAsync(useCallback(async () => getUserLocation(), [refreshState]))
+type useUserLocationReturn = {
+  data: UserLocationType | null
+  refresh: () => void
+  error: Error | null
+  loading: boolean
+}
 
-  const refresh = useCallback(() => {
-    setRefreshState(prev => !prev)
-  }, [])
+const useUserLocation = (): useUserLocationReturn => {
```

Comment:
:wrench: You can just do
```suggestion
const useUserLocation = (): Return<UserLocationType>
```